File: solve/0520_대충 만든 자판/BongSeon/postforty/대충 만든 자판.py
def solution(keymap, targets):
    def func(k, t):
        result = []
        for i in t:
            result.append(k.find(i) + 1 if k.find(i) >= 0 else -1)
        return result

    t_lst = []
    for t in targets:
        k_lst = []
        for k in keymap:
            k_lst.append(func(k, t))
        rst = 0
        for x in zip(*k_lst):
            # keymap으로 targets을 구현할 수 없는 경우
            if -1 in x and len(set(x)) == 1:
                rst = -1
                break
            # keymap으로 targets을 구현할 수 있는 경우 -1을 제외 후 최소 클릭 값 누적
            else:
                rst += min(filter(lambda n: n != -1, x))
        t_lst.append(rst)

    return t_lst


print(solution(["ABACD", "BCEFD"], ["ABCD", "AABB"]))  # [9, 4]
print(solution(["AA"], ["B"]))  # [-1]
print(solution(["BC"], ["AC", "BC"]))  # [-1,3]

File: solve/0520_대충 만든 자판/BongSeon/postforty/대충 만든 자판.js
function solution(keymap, targets) {
  function func(k, t) {
    const result = [];
    [...t].forEach((v, i) =>
      result.push(k.indexOf(v) >= 0 ? k.indexOf(v) + 1 : -1)
    );
    return result;
  }

  const tArr = [];
  targets.forEach((t) => {
    const kArr = keymap.map((k) => func(k, t));
    let rst = 0;
    // map 메소드를 이용한 파이썬의 for x in zip(*k_lst)와 유사한 코드
    const rstArr = kArr[0].map((_, index) => kArr.map((arr) => arr[index]));
    for (let x of rstArr) {
      if (x.includes(-1) && new Set(x).size === 1) {
        rst = -1;
        break;
      } else {
        rst += Math.min(...x.filter((el) => el !== -1));
      }
    }
    tArr.push(rst);
  });
  return tArr;
}

console.log(solution(["ABACD", "BCEFD"], ["ABCD", "AABB"])); // [9, 4]
console.log(solution(["AA"], ["B"])); // [-1]
console.log(solution(["BC"], ["AC", "BC"])); // [-1,3]
